devbuild.sh

```diff
@@ -136,6 +136,12 @@ while :; do
   shift
 done
 
+# Ensure platform name from variance ======================================
+if [[ "${PLATFORM}" == "wcoss2" || "${PLATFORM}" == "cactus" ||
+      "${PLATFORM}" == "dogwood" ]]; then
+  PLATFORM="wcoss2"
+fi
```

Comment:
I think it is better to live this to the user. This looks like we are going back to auto-detecting machine just like `detect_machine.sh` used to do. If the user is on 'cactus/dogwood' and wants to use wcoss2 modulefiles, he can just pretend to be on wcoss2 and pass `-p=wcoss2` instead.

Reply:
@danielabdi-noaa, the wcoss system is weird. It switches between `cactus` and `dogwood` on a regular basis. So both cactus and dogwood are called `wcoss2`. This if-statement prevent a user on wcoss2, either on cactus or on dogwood, from looking into this script or typing --help to figure out what is wrong.

Reply:
@chan-hoo But then you would have to make the same changes in `test/build.sh` as well. I still think it is better to keep it simple and let the user explicitly specify he wants to use wcoss2 modulefiles for the build. 
I was about to suggest that instead of changing `devbuild.sh` and `test/build.sh`, to provide modulefiles for `cactus/dogwood` that are sym-linked to `wcoss2` ones, but there is lots of code specific for wcoss2 `if [ platform = "wcoss" ]` that it probably complicates the code even more ...

Reply:
@danielabdi-noaa, I understand what your concern is. We might have a different viewpoint. I stand on a user's point of view, while you are on a developer's point of view :) :) If you feel uncomfortable about this statement, I'll remove it.

Reply:
@chan-hoo - I agree with @danielabdi-noaa, the user needs to be responsible for knowing which machine they are on and specify the correct machine.  The use of if-then-else clauses and checking for specific machines is an anti-pattern that leads to overly complicated hard-to-understand and unmaintainable code.  The names of machines should not appear anywhere in the UFS code or scripts, and I will not approve of changes that includes that.  Please remove this check.

Reply:
@christopherwharrop-noaa, okay, no problem. I'll remove it. My concern is that all users are not as smart as you expect :) :)

Reply:
@chan-hoo - Thank you for making that change.  One lesson I've learned the hard way is that making things stupid-proof is a fool's errand.  It's just not possible and causes more problems than it solves.  Knowing what machine you are using is a reasonable expectation.